managed/CounterStrikeSharp.API/Modules/Utils/StandaloneCommand.cs

```diff
@@ -17,36 +19,48 @@ public static void AddStandaloneCommand(string name, string description, Command
+                    // Do not execute command if we do not have the correct permissions.
+                    var permissions = methodInfo?.GetCustomAttributes<BaseRequiresPermissions>();
+                    if (permissions != null) permissionsToCheck.AddRange(permissions);
+                }
+                else
                 {
-                    foreach (var attr in permissions)
+                    var data = AdminManager.GetCommandOverrideData(name);
+                    if (data != null)
                     {
-                        attr.Command = name;
-                        if (!attr.CanExecuteCommand(caller))
-                        {
-                            command.ReplyToCommand("[CSS] You do not have the correct permissions to execute this command.");
-                            return;
-                        }
+                        var attrType = (data.CheckType == "all") ? typeof(RequiresPermissions) : typeof(RequiresPermissionsOr);
+                        var attr = (BaseRequiresPermissions)Activator.CreateInstance(attrType, args: AdminManager.GetPermissionOverrides(name));
+
+                        if (attr != null) permissionsToCheck.Add(attr);
                     }
                 }
-            }
-            // If this command has it's permissions overriden, we will do an AND check for all permissions.
-            else
-            {
-                // I don't know if this is the most sane implementation of this, can be edited in code review.
-                var data = AdminManager.GetCommandOverrideData(name);
-                if (data != null) 
+
+                foreach (var attr in permissionsToCheck)
                 {
-                    var attrType = (data.CheckType == "all") ? typeof(RequiresPermissions) : typeof(RequiresPermissionsOr);
-                    var attr = (BaseRequiresPermissions)Activator.CreateInstance(attrType, args: AdminManager.GetPermissionOverrides(name));
                     attr.Command = name;
                     if (!attr.CanExecuteCommand(caller))
                     {
-                        command.ReplyToCommand("[CSS] You do not have the correct permissions to execute this command.");
+                        if (attr.GetType() == typeof(RequiresPermissions))
+                        {
+                            var flags = attr.Permissions.Except(adminData?.GetAllFlags() ?? new HashSet<string>());
+                            flags = flags.Except(adminData?.Groups ?? new HashSet<string>());
+                            command.ReplyToCommand($"[CSS] You are missing the correct permissions ({string.Join(", ", flags)}) to execute this command.");
+                        }
+                        else
+                        {
+                            var flags = attr.Permissions.Except(adminData?.GetAllFlags() ?? new HashSet<string>());
+                            flags = flags.Except(adminData?.Groups ?? new HashSet<string>());
```

Comment:
Yup. It would be nice if we didn't have to copy/paste changes for both Standalone and Plugin commands. I thought I saw something that unified the two together in a separate PR?

Reply:
Yeah I did, that will go in after this one probably